src/components/icon/icon.spec.ts

```diff
@@ -251,6 +253,53 @@ describe('Icon broadcast service', () => {
 
       expect(events.length).to.equal(0);
     });
+
+    it('when multiple broadcast services are initialized they should not send sync events to each other.', async () => {
+      const collections = createIconDefaultMap<string, SvgIcon>();
+      const references = createIconDefaultMap<string, IconMeta>();
+
+      const evt1 = [];
+      const broadcast1 = new IconsStateBroadcast(collections, references);
+      (broadcast1 as any).channel.addEventListener(
+        'message',
+        (e: MessageEvent<BroadcastIconsChangeMessage>) => {
+          evt1.push(e);
+        }
+      );
+
+      const evt2 = [];
+      const broadcast2 = new IconsStateBroadcast(collections, references);
+      (broadcast2 as any).channel.addEventListener(
+        'message',
+        (e: MessageEvent<BroadcastIconsChangeMessage>) => {
+          evt2.push(e);
+        }
+      );
+
+      const evt3 = [];
+      const broadcast3 = new IconsStateBroadcast(collections, references);
+      (broadcast3 as any).channel.addEventListener(
+        'message',
+        (e: MessageEvent<BroadcastIconsChangeMessage>) => {
+          evt3.push(e);
+        }
+      );
+
+      // a peer is requesting a state sync
+      channel.postMessage({ actionType: ActionType.SyncState });
+      await nextFrame();
+
+      // all icon broadcasts must respond once on the channel with their state
+      // so 1 event from the peer + 1 response event with state per broadcast service.
+      expect(evt1.length).to.equal(4);
```

Comment:
Eh, actually it's the 1 from peer + 2 from the other two broadcast services + one from the global service since the previous tests populate the global icon registry (`getIconRegistry()`) which persists and has a broadcast service too :)
That also means this test will fail if ran first or alone with 'expected 3 to equal 4' so you might want to ensure the global service is always initialized as one of the test broadcasts by calling `getIconRegistry()`?

Reply:
But that doesn't make sense, does it? It's 3 new broadcast services, not 2. If we add the global one, it should make 5.

Reply:
It does, since the channel doesn't get its own messages - so with 3 services, each should get the replies from the 2 others + the initial request - so 3 in total :)